executor/client/client.go

```diff
@@ -47,30 +46,20 @@ func MakeClient(executorUrl string) *Client {
 	return c
 }
 
-func (c *Client) GetServiceForFunction(metadata *metav1.ObjectMeta) (string, error) {
+func (c *Client) GetServiceForFunction(metadata *metav1.ObjectMeta) (*http.Response, error) {
 	executorUrl := c.executorUrl + "/v2/getServiceForFunction"
 
 	body, err := json.Marshal(metadata)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	resp, err := http.Post(executorUrl, "application/json", bytes.NewReader(body))
 	if err != nil {
-		return "", err
```

Comment:
I think the client code should handle the HTTP response, and not return it out of the interface.  You could perhaps add a custom error type if you want to return more information in the error.